src/main/java/org/junitpioneer/jupiter/RetryingTestExtension.java

```diff
@@ -82,7 +84,34 @@ static FailedTestRetrier createFor(Method test) {
 			RetryingTest retryingTest = AnnotationSupport
 					.findAnnotation(test, RetryingTest.class)
 					.orElseThrow(() -> new IllegalStateException("@RetryingTest is missing."));
-			return new FailedTestRetrier(retryingTest.value());
+
+			int maxAttempts = retryingTest.maxAttempts();
+			String maxAttemptsField = "maxAttempts";
+			int minSuccess = retryingTest.minSuccess();
+
+			if (retryingTest.value() != 0) {
+				if (retryingTest.maxAttempts() != 0) {
+					throw new IllegalStateException("@RetryTest requires that one of `value` or `maxAttempts` be set, but not both.");
+				}
+
+				maxAttempts = retryingTest.value();
+				maxAttemptsField = "value";
+			}
+
+			if (maxAttempts == 0) {
+				// maxAttempts defaults to minSuccess
+				maxAttempts = minSuccess;
```

Comment:
At which point this becomes a `@RepeatedTest`. Why is this not an exception?

Reply:
I am not sure if I understand the question, but I will try to address it. With this condition, `@RepeatedTest(minSuccess = 4)` is equivalent to `@RepeatedTest(maxAttempts = 4, minSuccess = 4)`. So "the test must run 4 times successfully". Does that make sense?

Reply:
I completely misread your comment. You are correct. It becomes equivalent to a RepeatedTest. It can be an exception of you want it to. Someone could could still set maxAttempts == minSucess, but they would then have to do it explicitly. Do you want me to change it to an exception?

Reply:
Well, it's not exactly like RepeatedTest. `@RepeatedTest(4)` will always run the test 4 times, even if it fails. `RetryingTest(minSuccess = 4)` will run the test 4 times, but stop on the first failure. I do not know if that is useful or not. I am just pointing out the difference.

Reply:
The test suite will fail either way. Unless the test is *very* expensive, there is no real advantage to use `@RetryingTest` over `@RepeatedTest` in such a scenario. Am I missing something?

Reply:
Correct. My apologies. I was just pointing out the slight difference. I'm not arguing to keep it the way I have it.

just let me know what you want me to do. Do you want me to change to an exception?

Reply:
Sorry, I can't give you a definitive answer yet, because I am not sure. I'll let the maintainer team know about this, have a team discussion and update you with our decision.

Reply:
I'm more with @Michael1993, I think the max tries must always be higher than the min tries. Otherwise you don't need the annotation anyway as you can scale it down to one.

Reply:
It now raises an exception if maxAttempts is not set or if maxAttempts <= minSuccess.